### PR TITLE
fix: Gracefully handle empty root node for nested component

### PR DESCRIPTION
**Pull request by LinusBorg**
This PR covers a kind of edge case.

Conditions: 

1. The component to test is nested in an element (so the parent element is not the mount wrapper)
2. the component doesn't render anything (so `$el` is a comment node)
3. The component is extracted with `wrapper.findComponent(Component)`
4. The test tries to use `extractedWrapper.find()` or `ExtractedWrapper.findAll()`

This will fail with an error being thrown:
> `TypeError: this.element.querySelector(All) is not a function`

This PR handles this case in a graceful way.

### CODE REVIEW

**Comment by LinusBorg** [comment]
Huh, tests failed to run because of styles issues. Do I need to run prettier manually?

**Review by lmiller1990 — APPROVED** [review]
LGTM!

Yep, I just run prettier manually.

File: src/vueWrapper.ts [hunk 1]
```diff
@@ -223,7 +223,9 @@ export class VueWrapper<T extends ComponentPublicInstance>
   findAll(selector: string): DOMWrapper<Element>[] {
     const results = this.parentElement['__vue_app__']
       ? this.parentElement.querySelectorAll(selector)
-      : this.element.querySelectorAll(selector)
+      : this.element.querySelectorAll
+      ? this.element.querySelectorAll(selector)
+      : ([] as unknown as NodeListOf<Element>)
```
Comment (cexbrayat) [inline src/vueWrapper.ts:228]: nit: is this cast really necessary here? I would have thought that returning a simple array would be enough.

Comment (cexbrayat) [reply src/vueWrapper.ts:228]: I gave it a quick try, and it looks like it's unnecessary. I'll make a PR to remove it.

Comment (cexbrayat) [reply src/vueWrapper.ts:228]: My bad: it _was_ necessary!

Comment (LinusBorg) [reply src/vueWrapper.ts:228]: Yeah, tsd doesn't like it :-P
